exporter/exporterhelper/internal/persistent_queue.go

```diff
@@ -141,7 +141,7 @@ func (pq *persistentQueue[T]) initPersistentContiguousStorage(ctx context.Contex
 // Consume applies the provided function on the head of queue.
 // The call blocks until there is an item available or the queue is stopped.
 // The function returns true when an item is consumed or false if the queue is stopped.
-func (pq *persistentQueue[T]) Consume(consumeFunc func(context.Context, T)) bool {
+func (pq *persistentQueue[T]) Consume(consumeFunc func(context.Context, T) bool) bool {
```

Comment:
Per the same logic that we never call into this after stop you should remove stopChan from here as well.

Reply:
May not be possible this...